### PR TITLE
Document macro params for patterns in Nightshade

### PR DESCRIPTION
**Pull request by josephco**
Standardize text/content param as 'content'

### CODE REVIEW

**Comment by misscs** [comment]
Great docs! Really clear. Main piece of feedback is to pull in how jsdoc syntax for documenting optional and default params. This will help keep things more standard across js, sass, and macros. 

Use `[ ]` to denote an optional param. When a value is passed in, it sets the default

```
// Optional Param 
 @param {string} [somebody] - Somebody's name.

// Default value
@param {string} [somebody=John Doe] - Somebody's name.
```

Here are the docs for the param tag: [jsdoc](http://usejsdoc.org/tags-param.html). [sassdoc](http://sassdoc.com/annotations/#parameter) follows similar useage.

File: src/press_quote/quote.macros.html [hunk 1]
```diff
@@ -1,11 +1,16 @@
...
   * @overview Macros for press quotes.
+  * @module Press Quote
 #}
 
 {% from '../icons/icons.macros.html' import svg %}
 
 {#
   * Renders a single press quote with source logo beneath.
+  * @param {string} Quote text content
+  * @param {string} Associated press logo name
```
Comment (misscs) [inline src/press_quote/quote.macros.html]: Perhaps provide a little more information that this is what renders the svg logo

File: src/hero/hero.macros.html [hunk 2]
```diff
@@ -1,9 +1,18 @@
...
+  * @module Hero
 #}
 
 {#
   * Renders a hero or featured hero module.
+  * @param {string} Heading text content
+  * @param {string} Subheading text content
+  * @param {string} Background image imgix path
+  * @param {boolean} Setting to display a large featured hero (optional, default: false)
+  * @param {string} Class name for imgix fluid targeting (optional, default: 'imgix-fluid--nocss')
```
Comment (misscs) [inline src/hero/hero.macros.html]: I was confused by "Imgix fluid targeting". This is the class that folks can use if they need to apply additional imgix js manipulations, right? I would find it much clearer to state something along the aforementioned lines if that's the case.

File: src/buttons/buttons.macros.html [hunk 3]
```diff
@@ -1,9 +1,25 @@
...
 {% from '../icons/icons.macros.html' import icon as icon_macro %}
 
-{% macro button(text, style='primary', size=null, type='button', href=null, icon=null, class=null, id=null) %}
+{#
+  * Renders a Button
+  * @param {string} Button text content
+  * @param {string} Button style (optional, default: 'primary')
+  * @param {string} Button size (optional)
+  * @param {string} Button element type attribute (optional, default: 'button')
+  * @param {string} URL for button style links (optional)
```
Comment (misscs) [inline src/buttons/buttons.macros.html]: In this case it could be full URL or path

File: src/beacon/beacon.macros.html [hunk 4]
```diff
@@ -1,9 +1,13 @@
...
 #}
 
 {#
   * Renders a beacon/indicator button
+  * @param {string} Anchor path for beacon target (e.g. '#content')
+  * @param {string} Class name for style or JS targeting (optional)
+  * @param {string} ID for JS targeting (optional)
 #}
 {% macro beacon(href, class=null, id=null) %}
 <a class="beacon{% if class %} {{ class }}{% endif %}"{% if id %} id="{{ id }}"{% endif %} href="{{ href }}"></a>
```
Comment (misscs) [inline src/beacon/beacon.macros.html]: Probably clearer to go with either _beacon_ or _indicator_.  I vote for the former.

File: src/callout/callout.macros.html [hunk 5]
```diff
@@ -1,9 +1,13 @@
 {#
   * @overview Macros for the Callout module.
```
Comment (misscs) [inline src/callout/callout.macros.html:2]: For consistency, let's not mix title and lowercase in the overview.

File: src/blocks/blocks.macros.html [hunk 6]
```diff
@@ -1,11 +1,16 @@
 {#
   * @overview Macros for the Blocks module.
```
Comment (misscs) [inline src/blocks/blocks.macros.html:2]: For consistency, let's not mix title and lowercase in the overview.

File: src/gallery/gallery.macros.html [hunk 7]
```diff
@@ -1,9 +1,17 @@
 {#
   * @overview Macros for Gallery carousel that use Flickity
```
Comment (misscs) [inline src/gallery/gallery.macros.html]: For consistency, let's not mix title and lowercase in the overview.

File: src/callout/callout.macros.html (hunk 5 above)
Comment (josephco) [inline src/callout/callout.macros.html:2]: Capitalizing module names in all docs

File: src/beacon/beacon.macros.html (hunk 4 above)
Comment (josephco) [inline src/beacon/beacon.macros.html]: Done

File: src/hero/hero.macros.html (hunk 2 above)
Comment (josephco) [inline src/hero/hero.macros.html]: Reworded to be clearer

**Pull request merged**